app/models/tenant_quota.rb

```diff
@@ -111,4 +113,36 @@ def format
   def default_unit
     self.class.quota_definitions.fetch_path(name.to_sym, :unit).to_s
   end
+
+  def check_for_over_allocation
+    return unless value_changed?
+
+    # Root tenant has no (unlimited) quota
+    # First level tenant can also have unlimited quota
+    return if tenant.root? || tenant.parent.root?
+
+    if value_changed?
```

Comment:
You are already checking that value_changed? is true on the first line of this method.